startupscript/butane/aws/metadata-utils.sh

```diff
@@ -17,16 +17,27 @@ function get_metadata_value() {
   token=$(wget --method=PUT --header "X-aws-ec2-metadata-token-ttl-seconds:600" -q -O - http://169.254.169.254/latest/api/token)
   local id
   id=$(wget --header "X-aws-ec2-metadata-token: ${token}" -q -O - http://169.254.169.254/latest/meta-data/instance-id)
-  docker run --rm -d --network host \
+  docker run --rm --detach --network host \
     public.ecr.aws/aws-cli/aws-cli \
     ec2 describe-tags \
     --filters "Name=resource-id,Values=${id}" "Name=key,Values=${tag_key}" \
     --query "Tags[0].Value" --output text 2>/dev/null
 }
 readonly -f get_metadata_value 
 
+# guest attributes are not supported on EC2 instances. But to keep the interface consistent with GCP, we define a no-op function.
+function get_guest_attribute() {
+  get_metadata_value "$1"
+}
+readonly -f get_guest_attribute
+
+
 # Sets tags on the EC2 instance with the given key and value.
 function set_metadata() {
+  if [[ -z "$2" ]]; then
```

Comment:
What is the bad behavior if the value is empty?
The usage message on its own doesn't indicate that the value cannot be an empty string.

Reply:
actually you are right.. it can be empty